### PR TITLE
Feature/header navigation

### PR DESCRIPTION
**Pull request by DercilioFontes**
Header/Navigation #1

### CODE REVIEW

File: styles/Header.module.css [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    .menuModal {
+        position: fixed;
+        top: 0;
+        right: 0;
+        width: 80%;
+        background-color: #ffffff;
+    }
+
+    .menuHeader {
+        border-bottom: 1px solid #ccd2eb;
```
Comment (marceloglacial) [inline styles/Header.module.css:76]: please use css vars for the colors

**Review by marceloglacial — CHANGES_REQUESTED** [review]

File: components/FloatingActionButton/FloatingActionButton.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import styles from '@/styles/FloatingActionButton.module.css'
+import { Icon } from '@/components'
+import { useEffect, useState } from 'react'
+
+const FloatingActionButton = () => {
+    const [show, setShow] = useState(false)
+
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply components/FloatingActionButton/FloatingActionButton.jsx]: Done

File: components/FloatingActionButton/FloatingActionButton.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        return () => {
+            window.removeEventListener('scroll', scrollListener)
+        }
+    }, [])
+
+    return (
+        <>
+            {show ? (
+                <div
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply components/FloatingActionButton/FloatingActionButton.jsx]: Done

File: components/Header/Header.jsx [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+const Header = () => {
+    const [modalOpen, setModalOpen] = useState(false)
+
+    return (
+        <div className={styles.header + ' flex-row-space-between'}>
+            <Link href='/'>
+                <Image className={styles.logo} alt='logo small' src={smallLogo} />
+            </Link>
+            <div className={styles.empty}></div>
+            <Nav className={styles.nav + ' body1'} />
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply components/Header/Header.jsx]: Done

File: components/Icon/Icon.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import Image from 'next/image'
+import menu from '../../public/images/menu-icon.svg'
+import close from '../../public/images/close-icon.svg'
+import arrowUp from '../../public/images/arrow-up-icon.svg'
+
+const getSrc = (variant) => {
+    switch (variant) {
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply components/Icon/Icon.jsx]: Done

File: styles/Header.module.css [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    .logo {
+        width: 75%;
+    }
+
+    .nav {
+        display: none;
+    }
+
+    .navMobile a {
+        color: #383f60;
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply styles/Header.module.css]: Done

File: styles/Header.module.css [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        display: block;
+    }
+
+    .backdrop {
+        position: fixed;
+        top: 0;
+        right: 0;
+        height: 100vh;
+        width: 100%;
+        background-color: rgba(0, 0, 0, 0.5);
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply styles/Header.module.css]: Done

File: styles/Header.module.css [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        display: flex;
+        justify-content: flex-end;
+    }
+
+    .menuModal {
+        position: fixed;
+        top: 0;
+        right: 0;
+        width: 80%;
+        background-color: #ffffff;
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply styles/Header.module.css]: Done

File: styles/Header.module.css (hunk 1 above)
Comment (DercilioFontes) [reply styles/Header.module.css]: Done

File: components/FloatingActionButton/FloatingActionButton.jsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        window.addEventListener('scroll', scrollListener)
+
+        return () => {
+            window.removeEventListener('scroll', scrollListener)
+        }
+    }, [])
+
+    return (
+        <>
```
[the remark this answers is not in the document]
Comment (DercilioFontes) [reply components/FloatingActionButton/FloatingActionButton.jsx]: Done

File: components/Icon/Icon.jsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    arrowUp,
+}
+
+const Icon = (props) => {
+    const { variant, width, height } = props
+    const size = {}
+    if (width) size.width = width
+    if (height) size.height = height
+
+    return <Image {...size} alt={`${variant} icon`} src={icons[variant]} />
```
Comment (marceloglacial) [inline components/Icon/Icon.jsx:18]: One last thing, Icons usually are plain SVGs, not images

**Review by marceloglacial — CHANGES_REQUESTED** [review]

File: components/Icon/Icon.jsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+const icons = {
+    menu,
+    close,
+    arrowUp,
+}
+
+const Icon = (props) => {
+    const { variant, width, height } = props
+    const size = {}
```
Comment (marceloglacial) [inline components/Icon/Icon.jsx:14]: This could be: 
`size = { width, height }` 

No need to check if its exist

**Review by marceloglacial — CHANGES_REQUESTED** [review]

File: components/Icon/Icon.jsx (hunk 10 above)
Comment (DercilioFontes) [reply components/Icon/Icon.jsx]: Refactored

File: components/Icon/Icon.jsx (hunk 11 above)
Comment (DercilioFontes) [reply components/Icon/Icon.jsx]: Done

**Pull request merged**